docs/gene_sets.md

```diff
@@ -0,0 +1,122 @@
+
+When the portal successfully validates a gene set file, individual gene sets are extracted and stored in the database. If validation fails for an individual gene set in the file, then the entire operation fails. No gene sets from the file are stored in the database.
+
+The portal **does not** store the uploaded file. 
+
+The order of the gene symbols in each gene set is maintained in the database. (The order of gene symbols in the download for a specific gene set name matches the order of the gene symbols in the upload for a specific gene set name.)
+
+Multiple gene set files may be uploaded to the portal by the curator. 
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED" "MAY", and "OPTIONAL" in this document are to be interpreted as described in BCP 14, [RFC2119](https://www.rfc-editor.org/rfc/rfc2119.txt), and [RFC8174](https://www.rfc-editor.org/rfc/rfc8174.txt) when, and only when, they appear in all capitals, as shown here.
+
+## cellxgene gene set data format
+  
+The cellxgene gene set data format is a [*Tidy* CSV](./gene_sets_example.csv) (comma-separated values) file using UTF-8 encoding. Multiple gene sets MAY be included in the file similar to the [Gene Matrix Transposed](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMT:_Gene_Matrix_Transposed_file_format_.28.2A.gmt.29) or [Gene Matrix](https://software.broadinstitute.org/cancer/software/gsea/wiki/index.php/Data_formats#GMX:_Gene_MatriX_file_format_.28.2A.gmx.29) formats.
+
+
+
+Example:
+
+| gene_set_name | gene_set_description | gene_symbol | gene_description | provenance1      | provenance1_description |
+|---------------|----------------------|-------------|------------------|------------------|-------------------------|
+| club.cell     | description          | CCKAR       | description      | Pubmed ID XYZ123 | Primary Pubmed ID       |
+| club.cell     |                      | SCGB3A2     | description      | Pubmed ID ABC456 | Primary Pubmed ID       |
+| club.cell     |                      | CYP2F2      | description      | Pubmed ID DCF678 | Primary Pubmed ID       |
+| macrophage    | description          | CD68        | description      |                  |                         |
+| macrophage    |                      | CD163       | description      |                  |                         |
+
+
+## Mandatory Header
+
+The first row MUST contain header columns using reserved names in the following order:
+
+* `gene_set_name`
+* `gene_set_description`
+* `gene_symbol`
+* `gene_description`
+
+Publishers MAY include additional header columns. It is RECOMMENDED that these custom columns observe the same self-documenting style - `column_name` and `column_description` for easier comprehension by data consumers.
+
+Only the values of `gene_set_name` and `gene_set_description` are presented to data consumers in the portal UX. 
```

Comment:
@signechambers1 - do you want to add a sentence about the use of the values in cellxgene explorer?

Reply:
Sure, how about:

In the cellxgene explorer UX, users will be able to view `gene_set_name`, `gene_set_description` along with each `gene_symbol` and `gene_description` for the given gene set. Users will be able to color by the mean expression of the gene set, select cells from a histogram showing the distribution of mean expression, and plot gene sets on the scatter plot.